documentation/general/design/dash-holistic-view.md

```diff
@@ -0,0 +1,617 @@
+
+<figcaption><i>Figure 2 - DASH software stack</i></figcaption><br/>
+
+#### SDN controller
+
+The SDN controller is **primarily responsible for controlling the DASH overlay
+services**, while the traditional SONiC application containers are used to
+manage the underlay (L3 routing) and hardware platform. Both the DASH container
+and the traditional SONiC application containers sit atop the Switch State
+services (SWSS) layer, and manipulate the Redis application-layer DBs; these in
+turn are translated into SAI dataplane obects via the normal SONiC orchestration
+daemons inside SWSS.
+
+The SDN controller controls the overlay built on top of the physical layer
+(underlay) of the infrastructure.  From the point of view of the SDN control
+plane, when a customer creates an operation from the cloud portal (for example a
+VNET creation), the controller allocates the resources, placement management,
+capacity, etc. via the  **NorthBound interface APIs**.
+
+##### SDN and DPU High-Availability (HA)
+
+For **High Availability** (HA), the SDN controller selects a pair of cards and
+configures them identically.  The only requirement on the card from the HA
+perspective is for the cards to setup a channel between themselves for flow
+synchronization.  The synchronization mechanism is left for technology suppliers
+to define and implement. For more information, see [High Availability and
+Scale](../../high-avail/design/high-availability-and-scale.md) document.
+
+### DASH container
+
+The SDN controller communicates with a DASH device through a
+**[gNMI](https://github.com/Azure/DASH/wiki/Glossary#gnmi) endpoint** served by
+a new DASH SDN agent **running inside a new SONiC DASH container**.  
+
+In summary:
+
+- The DASH container translates SDN configuration modeled in gNMI into **SONiC
+  DB** objects. The gNMI schema is closely related to the DASH DB schema so in
+  effect, the gNMI server is a a thin RPC shim layer to the DB.
+- The **SONiC orchagent** inside the Switch State Service (SWSS) Container will
+  be enhanced to transform and translate these objects into **SAI_DB objects**,
+  including the new **DASH-specific SAI objects**.  
+- An **enhanced syncd** will then configure the dataplane using the **technology
+  supplier-specific SAI library**.
+
+A **gNMI schema** will manage the following DASH components:
+
+- Elastic Network Interface (ENI)
+- Access Control Lists (ACLs)
+- Routing and mappings
+- Encapsulations
+- Other  
+
+> [!NOTE] See also work in progress PR [doc-ref-config-example
+> #93](https://github.com/Azure/DASH/pull/93).
+
+#### Multiple DPUs device
+
+In the case of a multiple DPUs device the following applies:
+
+- Each DPU provides a gNMI endpoint for SDN controller through a unique IP
+  address.
+- An appliance or smart switch containing multiple DPUs will contain multiple
+  gNMI endpoints for SDN controller, and the controller treats each DPU as a
+  separate entity.
```

Comment:
Note:  Check w/SDN and SONiC that the controller communicates to the DPU, or is it the SONiC switch NOS communicating with the DPU?

Reply:
Added note to the content.